lib/src/arithmetic/multiplier.dart

```diff
@@ -38,89 +130,87 @@ abstract class Multiplier extends Module {
   /// multiplier [b]
   @protected
-  Logic? get selectSignedMultiplier => tryInput('selectSignedMultiplier');
+  Logic? get selectSignedMultiplier =>
+      signedMultiplierConfig?.tryRuntimeInput(this);
 
   /// Logic that tells us [product] is signed.
   @protected
   Logic get isProductSigned => output('isProductSigned');
 
-  /// Take input [a] and input [b] and return the
-  /// [product] of the multiplication result.
+  /// Take input [a] and input [b] and return the [product] of the
+  /// multiplication result.
   ///
-  /// [signedMultiplicand] parameter configures the multiplicand [a] as a signed
-  /// multiplier (default is unsigned).
+  /// The optional [signedMultiplicandParam] parameter configures the
+  /// multiplicand [a] as a signed multiplicand (default is unsigned) or with a
+  /// runtime configurable [selectSignedMultiplicand] input.
   ///
-  /// [signedMultiplier] parameter configures the multiplier [b] as a signed
-  /// multiplier (default is unsigned).
+  /// The optional [signedMultiplierParam] parameter configures the multiplier
+  /// [b] as a signed multiplier (default is unsigned) or with a runtime
+  /// configurable [selectSignedMultiplier] input.
   ///
-  /// Optional [selectSignedMultiplicand] allows for runtime configuration of
-  /// signed or unsigned operation, overriding the [signedMultiplicand] static
-  /// configuration.
-  ///
-  /// Optional [selectSignedMultiplier] allows for runtime configuration of
-  /// signed or unsigned operation, overriding the [signedMultiplier] static
-  /// configuration.
-  /// If [clk] is not null then a set of flops are used to make the multiply
-  /// a 2-cycle latency operation. [reset] and [enable] are optional
-  /// inputs to control these flops when [clk] is provided.
+  /// If [clk] is not null then a set of flops are used to make the multiply a
+  /// 2-cycle latency operation. [reset] and [enable] are optional inputs to
+  /// control these flops when [clk] is provided.
   Multiplier(Logic a, Logic b,
       {Logic? clk,
       Logic? reset,
       Logic? enable,
-      this.signedMultiplicand = false,
-      this.signedMultiplier = false,
-      Logic? selectSignedMultiplicand,
-      Logic? selectSignedMultiplier,
+      dynamic signedMultiplicandParam,
+      dynamic signedMultiplierParam,
       super.name = 'multiplier',
       String? definitionName})
       : super(
-            definitionName:
-                definitionName ?? 'Multiplier_W${a.width}x${b.width}') {
-    if (signedMultiplicand && (selectSignedMultiplicand != null)) {
-      throw RohdHclException('multiplicand sign reconfiguration requires '
-          'signedMultiplicand=false');
-    }
-    if (signedMultiplier && (selectSignedMultiplier != null)) {
-      throw RohdHclException('sign reconfiguration requires signed=false');
-    }
+            definitionName: definitionName ??
+                '${b.width}_$_signedMD(signedMultiplicandConfig)}'
```

Comment:
doesn't the naming here have `null` if it's null?  maybe something like 
```dart
[
  'asdf',
  if(x != null) '$x'
].nonNulls.join('_')
```

so that we can avoid null in the definition names and just omit things if they are not there?

but also, how is this not a "late variable not initialized" error at this point before you've set the config object?